rust/hyperlane-base/src/settings/parser/connection_parser.rs

```diff
@@ -0,0 +1,131 @@
+use eyre::eyre;
+use hyperlane_core::config::ConfigErrResultExt;
+use hyperlane_core::{config::ConfigParsingError, HyperlaneDomainProtocol};
+use url::Url;
+
+use crate::settings::envs::*;
+use crate::settings::ChainConnectionConf;
+
+use super::ValueParser;
+
+pub fn build_ethereum_connection_conf(
+    rpcs: &Vec<Url>,
+    chain: &ValueParser,
+    err: &mut ConfigParsingError,
+    default_rpc_consensus_type: &str,
+) -> Option<ChainConnectionConf> {
+    if rpcs.len() <= 1 {
```

Comment:
do we need to handle the case of it being len 0 where we'd I guess return None?

Reply:
we are returning `None` on len 0 here, because `next()` will return `None` and `map` doesn't apply the closure to `None`

Reply:
however I agree this implementation could be cleaner, there's an `unwrap` in the `else` branch which relies on the assumption that we're checking the length. Trying to make this nicer